### DISCUSSION TITLE
type_map

### DISCUSSION THREAD

**Discussion by foreverfreedown** [deepmodeling/deepmd-kit#852] [Q&A]
Since I generated two data sets and their type maps were opposite, I would like to know whether I can train directly regardless of the order of type map. If not, is there any good solution? Thank you for your help!

**Comment by njzjz** [comment]
Yes, the `data` key accepts a list and you can use multiple data sets.

**Reply by foreverfreedown** [reply]
Thank you for your reply. What I mean is that I can only specify the order of one type map during training, such as Ga N, while the type map of my data set has both Ga N and N Ga.  I want to ask them if the order before and after this will make a difference,and if so, how to solve it？

**Reply by njzjz** [reply]
The type maps should be consistent. In you case, both should be `Ga N`.

**Reply by njzjz** [reply]
You can use dpdata to resolve it:
```sh
pip install -U dpdata
```
```py
import dpdata
dpdata.MultiSystems(dpdata.LabeledSystem("system1", fmt="deepmd/npy"), dpdata.LabeledSystem("system2", fmt="deepmd/npy")).to_deepmd_npy("new_systems")
```
The new systems will locate at `new_systems` directory.

**Reply by foreverfreedown** [reply]
Thank you for your help!

**Reply by foreverfreedown** [reply]
I'm sorry that I still don't understand what you mean. What do you mean by system1 and system2 below? I still don't understand how to realize the change of two atoms in order.

**Reply by njzjz** [reply]
`system1` is the path to your data. (Take an example, https://github.com/deepmodeling/deepmd-kit/tree/master/examples/water/data)

**Reply by foreverfreedown** [reply]
I was wondering if it is possible to change 1 to 0, 0 to 1 directly in type.raw. Do you think it is feasible?

**Reply by njzjz** [reply]
Yes if you have provided `type_map.raw`. Otherwise you need to assign it manually.

**Reply by foreverfreedown** [reply]
Thank you for your help, I have provided type_map.raw, I think I could change directly to solve it.
